app/src/main/java/app/gamenative/PrefManager.kt

```
@@ -1265,6 +1265,22 @@ object PrefManager {
             setPref(CUSTOM_GAME_MANUAL_FOLDERS, Json.encodeToString(value))
         }
 
+    // Games the user has marked as favourite, stored as a set of LibraryItem.appId values so they
+    // work across every source (Steam, GOG, Epic, Amazon, custom games) without needing an account.
+    private val FAVOURITE_APP_IDS = stringPreferencesKey("favourite_app_ids")
+    var favouriteAppIds: Set<String>
+        get() {
+            val value = getPref(FAVOURITE_APP_IDS, "[]")
+            return try {
+                Json.decodeFromString<Set<String>>(value)
+            } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)